app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileActivity.java

```diff
@@ -0,0 +1,165 @@
+package com.nervousfish.nervousfish.activities;
+
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.v4.content.res.ResourcesCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.RadioButton;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.R;
+import com.nervousfish.nervousfish.data_objects.KeyPair;
+import com.nervousfish.nervousfish.data_objects.Profile;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import cn.pedant.SweetAlert.SweetAlertDialog;
+
+/**
+ * The main activity class that shows a list of all people with their public keys
+ */
+public final class CreateProfileActivity extends AppCompatActivity {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("CreateProfileActivity");
+    private static final int MIN_PASSWORD_LENGTH = 6;
+
+    private IServiceLocator serviceLocator;
+
+    /**
+     * Creates the new activity, should only be called by Android
+     *
+     * @param savedInstanceState Don't touch this
+     */
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.setContentView(R.layout.activity_create_profile);
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+
+        getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN);
+
+        LOGGER.info("CreateProfileActivity created");
+    }
+
+    /**
+     * Gets triggered when the Submit button is clicked.
+     *
+     * @param v The {@link View} clicked
+     */
+    public void onSubmitClick(final View v) {
+        if (validateInputFields()) {
+            final EditText nameInputField = (EditText) this.findViewById(R.id.profile_enter_name);
+            final String name = nameInputField.getText().toString();
+
+            final KeyPair keyPair = this.generateKeyPair();
+
+            try {
+                serviceLocator.getDatabase().addProfile(new Profile(name, keyPair));
+                new SweetAlertDialog(this, SweetAlertDialog.SUCCESS_TYPE)
```

Comment:
Can you split the 3 SweetAlertDialog up in 3 separate methods? BetterCodeHub doesn't like this long method